src/coreclr/vm/stackwalk.cpp

```diff
@@ -1546,6 +1546,70 @@ BOOL StackFrameIterator::IsValid(void)
     return TRUE;
 } // StackFrameIterator::IsValid()
 
+#ifndef DACCESS_COMPILE
+#ifdef FEATURE_EH_FUNCLETS
+//---------------------------------------------------------------------------------------
+//
+// Advance to the position that the other iterator is currently at.
+//
+void StackFrameIterator::SkipTo(StackFrameIterator *pOtherStackFrameIterator)
+{
+    // We copy the other stack frame iterator over the current one, but we need to
+    // keep a couple of members untouched. So we save them here and restore them
+    // after the copy.
+    ExInfo* pPrevExInfo = GetNextExInfo();
+    REGDISPLAY *pRD = m_crawl.GetRegisterSet();
+    GSCookie *pCurGSCookie = m_crawl.pCurGSCookie;
```

Comment:
Why do we need to patch GS cookie pointers?

I would expect that the GS cookie pointer moves in a lock step with registers, and so the one from `pOtherStackFrameIterator` should work fine.

Reply:
Maybe we don't, I just made it equivalent to what the loop that was there before was getting to. I'll check if we can keep the ones we've copied instead.

Reply:
You were right, it works fine without that too. I've added a commit to remove it.